webpack/assets/javascripts/react_app/components/ForemanModal/subcomponents/ForemanModalFooter.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Button } from 'patternfly-react';
+import { useModalContext } from '../ForemanModal.hooks';
+
+const ForemanModalFooter = props => {
+  const childCount = React.Children.count(props.children);
+  const { onClose } = useModalContext();
```

Comment:
Instead of reading the modal context, would it make more sense to just receive `onClose` as a prop?
You can achieve that by using `React.cloneElement` to clone the `footer` in the Modal component so you can pass your own `onClose` prop without using `context` API.
It might be useful when you will switch to redux.

Reply:
@sharvit I wanted to use context for this because I got the idea that it might be more useful in case the ForemanModal needs to get more complex and add more subcomponents/props/etc. in the future.  I followed the pattern I read about [here](https://www.jakewiesler.com/blog/compound-components-and-the-context-api/). The author lays out a good argument for using context instead of children-cloning.  (The examples use the old Context API, but this new one is even better because we can use hooks instead of render props.) And context and Redux aren't exclusive-- we can still connect the ForemanModal component to Redux.

Reply:
Make sense